mlrun/utils/helpers.py

```diff
@@ -670,19 +670,49 @@ def generate_artifact_uri(project, key, tag=None, iter=None):
     return artifact_uri
 
 
-def extend_hub_uri_if_needed(uri):
-    if not uri.startswith(hub_prefix):
-        return uri, False
-    name = uri[len(hub_prefix) :]
-    tag = "master"
-    if ":" in name:
-        loc = name.find(":")
-        tag = name[loc + 1 :]
-        name = name[:loc]
+def extend_hub_uri_if_needed(uri) -> Tuple[str, bool]:
+    """
+    Retrieve the full uri of the item's yaml in the hub.
+
+    :param uri: structure: "hub://[<source>/]<item-name>[:<version>]"
 
+    :return: (extended uri of item, is hub item)
+    """
+    is_hub_uri = uri.startswith(hub_prefix)
+    if not is_hub_uri:
+        return uri, is_hub_uri
+
+    db = mlrun.get_run_db()
+    name = uri.removeprefix(hub_prefix)
+    tag = "latest"
+    source_name = ""
+    if ":" in name:
+        name, tag = name.split(":")
+    if "/" in name:
+        try:
+            source_name, name = name.split("/")
+        except ValueError:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "Invalid character '/' in function name or source name"
+            )
     # hub function directory name are with underscores instead of hyphens
     name = name.replace("-", "_")
-    return config.get_hub_url().format(name=name, tag=tag), True
+    function_suffix = f"{name}/{tag}/src/function.yaml"
+    if not source_name:
+        # Searching item in all sources
+        sources = db.list_hub_sources(
+            item_name=normalize_name(name, verbose=False), tag=tag
```

Comment:
why normalize here and replace in line 699? you can normalize it once as I see it

Reply:
The normalize here is redundant, removing.